### ISSUE TITLE
[Accessibility]: Email sign up form input elements are not linked to their error states properly

### ISSUE THREAD

**Issue by kopancek** [sourcegraph/sourcegraph#35699]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34111

### Problem description

The sign up form contains various inputs (email, username, password) that can have different error states, e.g. if the email is invalid or username is already taken. The HTML input elements do not have `aria-describedby` attribute attached, which means that the error state might not be read correctly by a screen reader.

### Expected behavior

All 3 elements should have proper `aria-describedby` attribute linked to the id of the correct element that actually describes the error state

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.
